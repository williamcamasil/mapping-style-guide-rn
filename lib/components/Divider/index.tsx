import React from 'react';
import { View, ViewProps, ViewStyle } from 'react-native';

import { useViewStyles } from '../../hooks';
import { withTheme } from '../../theme';
import { AppThemeType } from '../../theme/default';
import { ColorPaletesType } from '../../tokens';

export type DividerPropsType = ViewProps & {
  theme: AppThemeType;
  color?: ColorPaletesType;
  size?: ViewStyle['height'];
  marginTop?: ViewStyle['margin'];
  marginBottom?: ViewStyle['margin'];
  marginLeft?: ViewStyle['margin'];
  marginRight?: ViewStyle['margin'];
};

const Divider: React.FC<DividerPropsType> = ({
  color,
  size,
  marginTop,
  marginBottom,
  marginLeft,
  marginRight,
  style,
  theme,
  ...others
}) => {
  const dividerStyles = useViewStyles(() => [
    {
      backgroundColor: theme.colors[color ?? 'neutralGray200'],
      height: size,
      marginTop,
      marginBottom,
      marginLeft,
      marginRight,
    },
    style,
  ], [
    theme.colors,
    color,
    marginBottom,
    marginLeft,
    marginRight,
    marginTop,
    size,
    style,
  ]);

  return (
    <View {...others} style={dividerStyles} />
  );
};

Divider.defaultProps = {
  size: 1,
};

export default withTheme(Divider);
