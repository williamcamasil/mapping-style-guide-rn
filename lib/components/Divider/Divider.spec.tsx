import React from 'react';

import { render } from '@testing-library/react-native';

import Divider from '.';

describe('Divider', () => {
  it('renders correctly', () => {
    render(
      <Divider />,
    );
  });
});

describe('Divider snapshot', () => {
  it('default', () => {
    const tree = render(
      <Divider />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('marings', () => {
    const tree = render(
      <Divider
        marginTop={1}
        marginLeft={2}
        marginBottom={3}
        marginRight={4}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
