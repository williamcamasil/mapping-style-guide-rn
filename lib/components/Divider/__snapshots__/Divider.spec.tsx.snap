// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider snapshot default 1`] = `
<View
  style={
    {
      "backgroundColor": "#F0F0F2",
      "height": 1,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
    }
  }
/>
`;

exports[`Divider snapshot marings 1`] = `
<View
  style={
    {
      "backgroundColor": "#F0F0F2",
      "height": 1,
      "marginBottom": 3,
      "marginLeft": 2,
      "marginRight": 4,
      "marginTop": 1,
    }
  }
/>
`;
